cdap-api-spark-base/src/main/java/io/cdap/cdap/api/spark/JavaSparkExecutionContextBase.java

```diff
@@ -49,9 +50,9 @@
  * Spark program execution context. User Spark program can interact with CDAP through this context.
  */
 @Beta
-public abstract class JavaSparkExecutionContextBase implements SchedulableProgramContext, RuntimeContext, Transactional,
-                                                               WorkflowInfoProvider, LineageRecorder,
-                                                               SecureStore, MetadataReader, MetadataWriter {
+public abstract class JavaSparkExecutionContextBase
+  implements SchedulableProgramContext, RuntimeContext, Transactional, WorkflowInfoProvider, LineageRecorder,
+  SecureStore, MetadataReader, MetadataWriter, FeatureFlagsProvider {
```

Comment:
You shouldn't need to add `FeatureFlagsProvider` since it is already coming from the `RuntimeContext`.